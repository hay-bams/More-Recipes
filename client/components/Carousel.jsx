import React from 'react';

const Carousel = () => (
  <div id="featured" className="carousel fade" data-ride="carousel">
    <ol className="carousel-indicators">
      <li data-target="#featured" data-slide-to="0" className="active" />
      <li data-target="#featured" data-slide-to="1" />
      <li data-target="#featured" data-slide-to="2" />
      <li data-target="#featured" data-slide-to="3" />
    </ol>

    <div className="carousel-inner fullheight">
      <div className="carousel-item active">
        <img
          className="d-block w-100"
          src="images/recipe26.jpg"
          alt="First slide"
        />
        <div className="carousel-caption carousel-caption-search">
          <h3 className="search-header">Find More Recipes</h3>
          <form className="form-inline my-2 my-lg-0">
            <input
              className="form-control form-control-lg mr-sm-2"
              type="text"
              placeholder="Search"
              aria-label="Search"
            />
            <button
              className="btn btn-success btn btn-lg my-2 my-sm-0"
              type="submit"
            >Search
            </button>
          </form>
        </div>
      </div>
      <div className="carousel-item">
        <img
          className="d-block w-100"
          src="images/recipe7.jpg"
          alt="Second slide"
        />
        <div className="carousel-caption carousel-caption-search">
          <h3 className="search-header">Find More Recipes</h3>
          <form className="form-inline my-2 my-lg-0">
            <input
              className="form-control form-control-lg mr-sm-2"
              type="text"
              placeholder="Search"
              aria-label="Search"
            />
            <button
              className="btn btn-success btn btn-lg my-2 my-sm-0"
              type="submit"
            >Search
            </button>
          </form>
        </div>
      </div>
      <div className="carousel-item">
        <img
          className="d-block w-100"
          src="images/recipe27.jpg"
          alt="Third slide"
        />
        <div className="carousel-caption carousel-caption-search">
          <h3 className="search-header">Find More Recipes</h3>
          <form className="form-inline my-2 my-lg-0">
            <input
              className="form-control form-control-lg mr-sm-2"
              type="text"
              placeholder="Search"
              aria-label="Search"
            />
            <button
              className="btn btn-success btn btn-lg my-2 my-sm-0"
              type="submit"
            >Search
            </button>
          </form>
        </div>
      </div>

      <div className="carousel-item">
        <img
          className="d-block w-100"
          src="images/recipe15.jpg"
          alt="Third slide"
        />
        <div className="carousel-caption carousel-caption-search">
          <h3 className="search-header">Find More Recipes</h3>
          <form className="form-inline my-2 my-lg-0">
            <input
              className="form-control form-control-lg mr-sm-2"
              type="text"
              placeholder="Search"
              aria-label="Search"
            />
            <button
              className="btn btn-success btn btn-lg my-2 my-sm-0"
              type="submit"
            >Search
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
);

export default Carousel;
