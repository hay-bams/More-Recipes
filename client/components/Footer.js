import React from 'react';

/**
 * @class Footer
 */
class Footer extends React.Component {
  render() {
    return (
      <footer>
        <div className = "container">
          <div className = "row">
            <div className = " col- col-sm-6">
                <p>Call us at <span className = "phone">+2348093747838</span></p>
                <p>All contents &copy; Ayobami</p>
            </div>

            <div className = "col- col-sm-6">
              <nav className="navbar navbar-expand navbar-dark ">
                  <ul className="navbar-nav ml-sm-auto">
                    <li className="nav-item active">
                      <a className="nav-link" href="add_recipe.html">Dashboard <span className="sr-only">(current)</span></a>
                    </li>
                    <li className="nav-item">
                      <a className="nav-link" href="catalogue.html">Catalogue</a>
                    </li>
                    <li className="nav-item">
                      <a className="nav-link" href="register.html">Signup</a>
                    </li>
                  </ul>
              </nav> 
            </div>
          </div>
        </div> 
      </footer>
    )
  }
}

export default Footer;