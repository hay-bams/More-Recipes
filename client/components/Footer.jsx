import React from 'react';

const Footer = () => (
  <footer>
    <div className="container">
      <div className="row">
        <div className=" col- col-sm-12">
          {/* <p className="text-center">Call us at <span className="phone">+2348093747838</span></p> */}
          <p className="text-center">All contents &copy; Ayobami</p>
        </div>

        <div className="col- col-sm-6">
          <nav className="navbar navbar-expand navbar-dark" />
        </div>
      </div>
    </div>
  </footer>
);

export default Footer;
