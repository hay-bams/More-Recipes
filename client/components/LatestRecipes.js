import React from 'react';
import { Link } from 'react-router-dom';

/**
 * @class LatestRecipes
 */
class LatestRecipes extends React.Component {
  render() {
    return (
      <div className = "container">
        <div className = "row">
          <div className = "col-sm">
            <h2 className = "text-center latest">Latest Recipes</h2>
          </div>
        </div>

        <div className = "row">
          <div className = "col-12 col-sm-6 col-md-4 col-lg-3 recipes">
            <div className="card recipe-card recipe-card" style={{border: 'none'}}>
              <img className="card-img-top img-fluid" src="images/recipe5.jpg" alt="Card image cap" style = {{height: 200 + 'px'}} />
              <div className="card-body">
                <h4 className="card-title">Recipe name</h4>
                <p className="card-text">
                  <span className = "row">
                    <a href="#" class = "text-success"><i class="fa fa-thumbs-up col-4" aria-hidden="true"></i>10</a>
                    <a href="#" class = "text-info"><i class="fa fa-comment col-4" aria-hidden="true"></i>25</a>
                    <a href="#" class = "text-danger"><i class="fa fa-thumbs-down col-4" aria-hidden="true"></i>3</a>
                  </span>
                </p>
                <Link to='/details' className="btn btn-info">view details</Link>
              </div>
            </div>
          </div>
          <div className = "col-12 col-sm-6 col-md-4 col-lg-3 recipes">
            <div className="card recipe-card" style = {{border: 'none'}}>
              <img className="card-img-top img-fluid" src="images/recipe2.jpg" alt="Card image cap" style = {{height: 200 + 'px'}} />
              <div className="card-body">
                <h4 className="card-title">Recipe name</h4>
                <p className="card-text">
                  <span className = "row">
                    <a href="#" className = "text-success"><i className="fa fa-thumbs-up col-6" aria-hidden="true"></i>10</a>
                    <a href="#" className = "text-info"><i className="fa fa-comment col-6" aria-hidden="true"></i>25</a>
                    <a href="#" className = "text-danger"><i className="fa fa-thumbs-down col-6" aria-hidden="true"></i>3</a>
                  </span>
                </p>
                <Link to='/details' className="btn btn-info">view details</Link>
              </div>
            </div>
          </div>

          <div className = "col-12 col-sm-6 col-md-4 col-lg-3 recipes">
            <div className="card recipe-card" style = {{border: 'none'}}>
              <img className="card-img-top img-fluid" src="images/recipe3.jpg" alt="Card image cap" style = {{height: 200 + 'px'}} />
              <div className="card-body">
                <h4 className="card-title">Recipe name</h4>
                <p className="card-text">
                  <span className = "row">
                    <a href="#" className = "text-success"><i className="fa fa-thumbs-up col-6" aria-hidden="true"></i>10</a>
                    <a href="#" className = "text-info"><i className="fa fa-comment col-6" aria-hidden="true"></i>25</a>
                    <a href="#" className = "text-danger"><i className="fa fa-thumbs-down col-6" aria-hidden="true"></i>3</a>
                  </span>
                </p>
                <Link to='/details' className="btn btn-info">view details</Link>
              </div>
            </div>
          </div>

          <div className = "col-12 col-sm-6 col-md-4 col-lg-3 recipes d-md-none d-lg-block">
            <div className="card recipe-card" style = {{border: 'none'}}>
              <img className="card-img-top img-fluid" src="images/recipe4.jpg" alt="Card image cap" style = {{height: 200 + 'px'}} />
              <div className="card-body">
                <h4 className="card-title">Recipe name</h4>
                <p className="card-text">
                  <span className = "row">
                    <a href="#" className = "text-success"><i className="fa fa-thumbs-up col-6" aria-hidden="true"></i>10</a>
                    <a href="#" className = "text-info"><i className="fa fa-comment col-6" aria-hidden="true"></i>25</a>
                    <a href="#" className = "text-danger"><i className="fa fa-thumbs-down col-6" aria-hidden="true"></i>3</a>
                  </span>
                </p>
                <Link to='/details' className="btn btn-info">view details</Link>
              </div>
            </div>
          </div>
        </div>
      </div>
    )
  }
}

export default LatestRecipes;