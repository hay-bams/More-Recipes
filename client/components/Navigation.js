import React from 'react';
import {Link, NavLink} from 'react-router-dom';

class Navigation extends React.Component {
    render() {
      return (
        <header>
        <nav className="navbar navbar-expand-md navbar-light fixed-top navigationBar">
          <div className = "container">
            <NavLink to='/' className="navbar-brand">My Recipes</NavLink>
            <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
              <span className="navbar-toggler-icon"></span>
            </button>
  
            <div className="collapse navbar-collapse resetNav" id="navbarSupportedContent">
              <ul className="navbar-nav ml-auto">
                <li className = "nav-item active">
                </li>
                <li className="nav-item active">
                  <NavLink to='/signin' className="nav-link">Sign in <span className="sr-only">(current)</span></NavLink>
                </li>
                <li className="nav-item">
                  <NavLink to='/signup' className="nav-link">Sign up</NavLink>
                </li>
              </ul>
            </div>
            </div>
        </nav>
      </header>
      )
    }
}

export default Navigation;