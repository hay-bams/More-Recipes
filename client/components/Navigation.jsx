import React from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

const Navigation = props => (
  <header>
    <nav
      className="navbar navbar-expand-md navbar-light fixed-top navigationBar"
    >
      <div className="container">
        <NavLink to="/" className="navbar-brand">My Recipes</NavLink>
        <button
          className="navbar-toggler"
          type="button"
          data-toggle="collapse"
          data-target="#navbarSupportedContent"
          aria-controls="navbarSupportedContent"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span className="navbar-toggler-icon" />
        </button>

        <div
          className="collapse navbar-collapse resetNav"
          id="navbarSupportedContent"
        >
          <ul className="navbar-nav ml-auto">
            <li className="nav-item active" />

            <li className="nav-item active">
              { props.userData === undefined ?
                <NavLink
                  to="/signin"
                  className="nav-link"
                  href="login.html"
                >Sign in
                  <span className="sr-only">(current)</span>
                </NavLink> : ''
              }
            </li>

            <li className="nav-item">
              { props.userData === undefined ?
                <NavLink
                  to="/signup"
                  className="nav-link"
                  href="register.html"
                >Sign up
                </NavLink> : ''
              }
            </li>

          </ul>
        </div>
      </div>
    </nav>
  </header>
);

const mapStateToProps = state => ({
  userData: state.userData
});

Navigation.defaultProps = {
  userData: {}
};

Navigation.propTypes = {
  userData: PropTypes.shape({
    user: PropTypes.shape({
      email: PropTypes.string,
      firstName: PropTypes.string,
      id: PropTypes.number,
      lastName: PropTypes.string
    })
  })
};

export default connect(mapStateToProps, null)(Navigation);
