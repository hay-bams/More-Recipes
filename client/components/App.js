import React from 'react';
import HomePageheader from './homePageComponents/HomePageHeader';
import LatestRecipes from './LatestRecipes';
import PopularRecipes from './PopularRecipes';
import Footer from './Footer';


/**
 * @class App
 */
class App extends React.Component {    
    render() {
        return (
          <div>
            <HomePageheader />
            <LatestRecipes />
            <PopularRecipes />
            <Footer />
          </div>
        )
    }    
}

export default App;