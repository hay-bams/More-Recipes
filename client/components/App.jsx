import React from 'react';
import HomePageheader from './homePageComponents/HomePageHeader';
import LatestRecipes from './LatestRecipes';
import PopularRecipes from './PopularRecipes';
import Footer from './Footer';

const App = () => (
  <div>
    <HomePageheader />
    <LatestRecipes />
    <PopularRecipes />
    <Footer />
  </div>
);

export default App;
