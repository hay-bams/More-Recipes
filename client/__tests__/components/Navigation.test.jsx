import React from 'react';
import { shallow } from 'enzyme';
import { Navigation } from '../../components/Navigation';
// import toJSON from 'enzyme-to-json';

describe('Navigation', () => {
  test('should render Navigation correctly', () => {
    const wrapper = shallow(<Navigation location={{}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
