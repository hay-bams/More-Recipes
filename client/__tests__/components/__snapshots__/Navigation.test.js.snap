// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should render Navigation correctly 1`] = `
<header>
  <nav
    className="navbar navbar-expand-md navbar-light fixed-top navigationBar"
  >
    <div
      className="container-fluid"
    >
      <NavLink
        activeClassName="active"
        ariaCurrent="true"
        className="navbar-brand"
        to="/"
      >
        More Recipes
      </NavLink>
      <button
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-target="#navbarSupportedContent"
        data-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse resetNav"
        id="navbarSupportedContent"
      >
        <ul
          className="navbar-nav ml-auto"
        >
          <li
            className="nav-item"
          >
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav-link"
              to="/catalogue"
            >
              Catalogue
            </NavLink>
          </li>
          <li
            className="nav-item active"
          >
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav-link"
              href="login.html"
              to="/signin"
            >
              Sign in
              <span
                className="sr-only"
              >
                (current)
              </span>
            </NavLink>
          </li>
          <li
            className="nav-item"
          >
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav-link"
              href="register.html"
              to="/signup"
            >
              Sign up
            </NavLink>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</header>
`;
